File: Algorithm/src/edu/Q2108.java
package edu;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

public class Q2108 {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int n;
		int c = 0;
		int sum = 0;
		int no1;
		int no2, no3, no4;
		int max;
		int index;
		
		n = Integer.parseInt(br.readLine());
		int[] num = new int[n];
		int[] count = new int[n];
		
		
		Arrays.fill(count, 1);
		
		while(c<n) {
			num[c] = Integer.parseInt(br.readLine());
			sum = sum + num[c];
			c = c + 1;
		}
		
		Arrays.sort(num);
		
		for(int i = 1; i < n; i++) {
			if(num[i] == num[i-1]) {
				count[i] = count[i] + count[i-1];
			}
		}
		
		max = count[0];
		index = 0;
		c = 1;
		
		for(int i = 1; i < n; i++) {
			if(max<count[i]) {
				max = count[i];
				index = i;
				c = 1;
			}else if(max == count[i]) {
				c = c + 1;
				if(c == 2) {
					max = count[i];
					index = i;
				}
			}
		}
		
		no1 = (int)Math.round( (float)sum / (float)n);
		no2 = num[(n-1)/2];
		no3 = num[index];
		no4 = num[n-1] - num[0];
		
		bw.write(Integer.toString(no1) + "\n");
		bw.write(Integer.toString(no2) + "\n");
		bw.write(Integer.toString(no3) + "\n");
		bw.write(Integer.toString(no4));
		bw.close();
	}

}

File: to python/Q2108.py
import sys

if __name__ == "__main__":

    m = int(sys.stdin.readline())
    arr1 = list()
    dic = list()
    sum = 0

    for i in range(m):
        n = int(sys.stdin.readline())
        arr1.append(n)
        sum += n

    dic = [[i, arr1.count(i)] for i in set(arr1)]

    arr1.sort()

    dic.sort(key = lambda x: x[1],reverse = True)
    
    print(round(sum/m))
    print(arr1[int(m/2)])
    if len(dic) == 1 or dic[0][1] != dic[1][1]:
        print(dic[0][0])
    else:
        print(dic[1][0])
    print(arr1[m-1] - arr1[0])
